command/s3_get.go

```diff
@@ -229,20 +226,34 @@ func (c *s3get) get(ctx context.Context) error {
 			}
 		}
 
-		// open the local file
-		file, err := os.Create(c.LocalFile)
-		if err != nil {
-			return errors.Wrapf(err, "error opening local file %v", c.LocalFile)
-		}
-		defer file.Close()
+		// download to local file
+		return errors.Wrapf(c.bucket.Download(ctx, c.RemoteFile, c.LocalFile),
+			"error downloading %s to %s", c.RemoteFile, c.LocalFile)
 
-		_, err = io.Copy(file, reader)
-		return errors.WithStack(err)
 	}
 
+	reader, err := c.bucket.Reader(ctx, c.RemoteFile)
+	if err != nil {
+		return errors.WithStack(err)
+	}
 	if err := util.ExtractTarball(ctx, reader, c.ExtractTo, []string{}); err != nil {
 		return errors.Wrapf(err, "problem extracting %s from archive", c.RemoteFile)
 	}
 
 	return nil
 }
+
+func (c *s3get) createPailBucket() error {
+	opts := pail.S3Options{
+		Credentials: credentials.NewCredentials(&credentials.StaticProvider{
+			Value: credentials.Value{
+				AccessKeyID:     c.AwsKey,
+				SecretAccessKey: c.AwsSecret,
+			}}),
```

Comment:
Can we do this using the new pail structures so we don't need to reference the credentials structure?